src/palace/manager/metadata_layer/metadata.py

```diff
@@ -27,124 +30,76 @@
-        else:
-            self.data_source_obj = None
-            self.data_source_name = self._data_source
-
-        self.title = title
-        self.sort_title = sort_title
-        self.subtitle = subtitle
-        if language:
-            language = LanguageCodes.string_to_alpha_3(language)
-        self.language = language
-        # medium is book/audio/video, etc.
-        self.medium = medium
-        self.series = series
-        self.series_position = series_position
-        self.publisher = publisher
-        self.imprint = imprint
-        self.issued = issued
-        self.published = published
-        self.duration = duration
-
-        if isinstance(primary_identifier, Identifier):
-            primary_identifier = IdentifierData(
-                type=primary_identifier.type, identifier=primary_identifier.identifier
-            )
-        self.primary_identifier = primary_identifier
-        self.identifiers = identifiers or []
-        self.permanent_work_id: str | None = None
-        if self.primary_identifier and self.primary_identifier not in self.identifiers:
-            self.identifiers.append(self.primary_identifier)
-        self.subjects = subjects or []
-        self.contributors = contributors or []
-        self.measurements = measurements or []
-
-        self.circulation = circulation
-
-        # renamed last_update_time to data_source_last_updated
-        self.data_source_last_updated = data_source_last_updated
-
-        self.__links: list[LinkData] = []
-        # The type ignore here is necessary because mypy does not like when a property setter and
-        # getter have different types. A PR just went in to fix this in mypy, so this should be able
-        # to be removed once mypy 1.16 is released.
-        # See: https://github.com/python/mypy/pull/18510
-        self.links = links  # type: ignore[assignment]
-
-    @property
-    def links(self) -> list[LinkData]:
-        return self.__links
+    title: str | None = None
+    subtitle: str | None = None
+    sort_title: str | None = None
+    language: str | None = None
+    medium: str | None = None
+    series: str | None = None
+    series_position: int | None = None
+    publisher: str | None = None
+    imprint: str | None = None
+    issued: datetime.date | None = None
+    published: datetime.date | None = None
+    identifiers: list[IdentifierData] = Field(default_factory=list)
+    subjects: list[SubjectData] = Field(default_factory=list)
+    contributors: list[ContributorData] = Field(default_factory=list)
+    measurements: list[MeasurementData] = Field(default_factory=list)
+    links: list[LinkData] = Field(default_factory=list)
+    data_source_last_updated: datetime.datetime | None = None
+    duration: float | None = None
+    permanent_work_id: str | None = None
+    # Note: brought back to keep callers of bibliographic extraction process_one() methods simple.
```

Comment:
This is what I was talking about above:  if there were a way to decouple CirculationData from the Metadata I _think_ (maybe?) it would be  easier to understand the model.